### PR TITLE
pre-commit: Add a new hook to run `isort`

### PR DESCRIPTION
**Pull request by xdustinface**
Inspired by https://github.com/Chia-Network/chia-blockchain/pull/8821#discussion_r729079099

### CODE REVIEW

**Review by jack60612 — DISMISSED** [review]
Lgtm

**Review by altendky — APPROVED** [review]
Looks enough like https://github.com/twisted/twisted/blob/0e54ecabe83e2f2dc75a9ac6c765e891e2f9381e/.pre-commit-config.yaml#L8-L11 for me.  I see it did in fact run in CI and pass at https://github.com/Chia-Network/chia-blockchain/pull/8827/checks?check_run_id=3896686769#step:4:81.

I don't know etiquette related to imposing new rules here or for massive (even if not major) changes like this so I would expect some discussion or somesuch.  But, it's good for me.

Maybe it would be nice to ban relative imports?  We have a few scattered around and it seems to me the sort of thing you generally ought to either always do or never do but this could make sense as a separate PR anyways.

**Comment by hoffmang9** [comment]
In the past it's been me making these sorts of changes. I'd generally post on Keybase soliciting any objections. One note is that we need to update CONTRIBUTING to add this point.

**Review by altendky — CHANGES_REQUESTED** [review]
To formalize @hoffmang9's comment, here's a change request.

**Review by altendky — DISMISSED** [review]

**Comment by xdustinface** [comment]
@altendky I added added the workaround for the pre-commit-isort-install issue, see 76907da4279b75377a0e83d9e37d617de3a810b3 + your commits to exclude the currently unsorted files + i rebased on `main` and adjusted the ignore list in 71d6e3a.

File: setup.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "pytest-asyncio",
     "pytest-monitor; sys_platform == 'linux'",
     "pytest-xdist",
+    "isort",
```
Comment (altendky) [inline setup.py:48]: Apparently we need a dsort too...  `:]`  (dependency sort)

**Review by altendky — DISMISSED** [review]
Close.  I should have remembered to add https://github.com/xdustinface/chia-blockchain/pull/3 before.  Basically, clear the ignore list, run that command, copy paste back in as the ignore list to update it.

Thanks for keeping at this.

File: setup.py (hunk 1 above)
Comment (altendky) [reply setup.py:48]: (no, don't change anything here.  just noting pre-existing unsorted deps...)

**Comment by xdustinface** [comment]
I regenerated the list already, again see 71d6e3a in case you missed it in my last comment. About adding that command as comment, Why should we ever need to regenerate that list? All i want to see there is entries getting dropped tbh 😄

**Review by altendky — DISMISSED** [review]
Thanks again!

**Comment by xdustinface** [comment]
Rebased

**Review by altendky — DISMISSED** [review]

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**